packages/gatsby-dev-cli/src/local-npm-registry/check-deps-changes.js

```diff
@@ -0,0 +1,190 @@
+const fs = require(`fs-extra`)
+const _ = require(`lodash`)
+const { getMonorepoPackageJsonPath } = require(`./utils`)
+const request = require(`request`)
+
+function difference(object, base) {
+  function changes(object, base) {
+    return _.transform(object, function(result, value, key) {
+      if (!_.isEqual(value, base[key])) {
+        result[key] =
+          _.isObject(value) && _.isObject(base[key])
+            ? changes(value, base[key])
+            : value
+      }
+    })
+  }
+  return changes(object, base)
+}
+
+/**
+ * Compare dependencies of installed packages and monorepo package.
+ * It will skip dependencies that are removed in monorepo package.
+ *
+ * If local package is not installed, it will check unpkg.com.
+ * This allow gatsby-dev to skip publishing unnecesairly and
+ * let install packages from public npm repository if nothing changed.
+ */
+const checkDepsChanges = async ({
+  newPath,
+  packageName,
+  monoRepoPackages,
+  root,
+  isInitialScan,
+  ignoredPackageJSON,
+}) => {
+  const monoRepoPackageJsonPath = getMonorepoPackageJsonPath({
+    packageName,
+    root,
+  })
+  const monorepoPKGjsonString = fs.readFileSync(
+    monoRepoPackageJsonPath,
+    `utf-8`
+  )
+  const monorepoPKGjson = JSON.parse(monorepoPKGjsonString)
+  if (ignoredPackageJSON.has(packageName)) {
+    if (ignoredPackageJSON.get(packageName).includes(monorepoPKGjsonString)) {
+      // we are in middle of publishing and content of package.json is one set during publish process,
+      // so we need to not cause false positives
+      return {
+        didDepsChanged: false,
+      }
+    }
+  }
+
+  let localPKGjson
+  let packageNotInstalled = false
+  try {
+    localPKGjson = JSON.parse(fs.readFileSync(newPath))
+  } catch {
+    packageNotInstalled = true
+    // there is no local package - so we still need to install deps
+    // this is nice because devs won't need to do initial package installation - we can handle this.
+    if (isInitialScan) {
```

Comment:
we could rewrite this one into the following. Might be a bit better to read
```
  if (!isInitialScan) {
    console.log(
      `'${packageName}' doesn't seem to be installed. Restart gatsby-dev to publish it`
    )
    return {
      didDepsChanged: false,
      packageNotInstalled,
    }
  }

  // if package is not installed, we will do http GET request to
  // unkpg to check if dependency in package published in public
  // npm repository are different

  // this allow us to not publish to local repository
  // and save some time/work
  try {
    localPKGjson = await new Promise((resolve, reject) => {
      request(`https://unpkg.com/${packageName}/package.json`, function (
        error,
        response,
        body
      ) {
        if (response && response.statusCode === 200) {
          return resolve(JSON.parse(body))
        }

        return reject(error)
      })
    })
  } catch {
    console.log(
      `'${packageName}' doesn't seem to be installed and is not published on NPM.`
    )
    return {
      didDepsChanged: true,
      packageNotInstalled,
    }
  }
```

Reply:
done!